### PR TITLE
Feature mandatory fields

### PR DESCRIPTION
**Pull request by jjchee77**
Resolves

- #54 
- #75 

Updates the `value` field of the `Email` and `Address` model to be optional, with a default empty string being used to represent the value if it is not supplied by the user. Note that this does not remove the associated validation checks for the `value` field. Validation check is updated to ignore empty strings.

Old test cases written to ensure the presence of the now optional `value` field for `Email` and `Address`, has been updated to reflect the expected outcome.

Users can now add a new person simply by supplying a name and phone number.

**Edit**: The current decision to use an empty string instead of `java.util.Optional` is due to the unnecessary complexity involved in introducing the alternative design at this stage. By default, the existing `com.fasterxml.jackson` package library serializes an `Optional` field as an object with  a `present` boolean field instead of the value itself, and does not support the instantiation of `Optional` fields upon deserialization. However, there appears to be an existing library that already supports this and will be considered in the next mini-iteration.

### CODE REVIEW

**Review by dillontkh — COMMENTED** [review]
Could you add a comment explaining the field cannot be Optional thing in the relevant places?

Apart from that, I pulled your PR and tested the app, LGTM!

**Comment by jjchee77** [comment]
> Could you add a comment explaining the field cannot be Optional thing in the relevant places?
> 
> Apart from that, I pulled your PR and tested the app, LGTM!

Edited PR details!

**Comment by dillontkh** [comment]
> > Could you add a comment explaining the field cannot be Optional thing in the relevant places?
> > Apart from that, I pulled your PR and tested the app, LGTM!
> 
> Edited PR details!

I meant comments in the code so future devs won't try and make the same mistake.

**Comment by jjchee77** [comment]
Sorry just to clarify, are u referring to adding comments to indicate all the areas of the code that has to be modified if the `value` field in the `Email` and `Address` model were to be changed from type `String` to `Optional<String>`?

If so, where would such an a comment be appropriate and to what extent should it be detailed? I restored the stash for this previous attempt and 15 files were changed.

**Comment by dillontkh** [comment]
Hm then perhaps the DG would be a better place to document this?

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 1]
```diff
@@ -58,5 +58,14 @@ public PersonCard(Person person, int displayedIndex) {
                 .sorted(Comparator.comparing(tag -> tag.tagName))
                 .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
         note.setText(person.getNote().value);
+
+        // Bind manageability (presence) of node based on presence of value for optional fields
+        address.setVisible(!person.getAddress().value.isEmpty());
+        email.setVisible(!person.getEmail().value.isEmpty());
+        note.setVisible(!person.getNote().value.isEmpty());
```
Comment (ruijietay) [inline src/main/java/seedu/address/ui/PersonCard.java:65]: Thank you for modifying the visibility of the note too!

**Review by ruijietay — APPROVED** [review]
I've pulled this PR and tested it locally, great work!

As for the documenting of the usage of an empty string over an Optional, I also think it would be good to include it in the DG.

Other than that, LGTM 🥳

**Comment by jjchee77** [comment]
Ok will update the DG accordingly in iteration v1.2b once I finalize the approach to implementing an optional field for a client.

**Comment by dillontkh** [comment]
Alright sounds good

**Review by dillontkh — APPROVED** [review]
LGTM!

**Pull request merged**